core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -165,20 +156,42 @@ class RecordInsightsLOCO[T <: Model[T]]
 
   private def returnTopPosNeg
   (
-    featureArray: Array[(Int, Double)],
+    featureSparse: SparseVector,
+    zeroValIndices: Array[Int],
     featureSize: Int,
     baseScore: Array[Double],
     k: Int,
     indexToExamine: Int
   ): Seq[LOCOValue] = {
-
     val minMaxHeap = new MinMaxHeap(k)
     val aggregationMap = mutable.Map.empty[String, (Array[Int], Array[Double])]
-    for {i <- featureArray.indices} {
-      val (oldInd, oldVal) = featureArray(i)
-      val diffToExamine = computeDiffs(i, oldInd, oldVal, featureArray, featureSize, baseScore)
-      val history = histories(oldInd)
 
+    agggregateDiffs(0, Left(featureSparse), indexToExamine, minMaxHeap, aggregationMap,
+      baseScore)
```

Comment:
So the sparse features you just put in a value of 0? Cant we just skip adding them to the heap?

Reply:
I had the same idea but in one of the iteration I ran into test failures and deferred it to later. I'll recheck now that I have everything green

Reply:
What kind of failures have you encountered?

Reply:
it may be that we were doing an unnecessary calculation and that just happened to be captured in the test...

Reply:
@michaelweilsalesforce you can reproduce it by commenting out the line 171-172.
```
Aggregate all the derived hashing tf features of rawFeature - text. 0.08025355373244505 was not less than 1.0E-10 expected aggregated LOCO value (0.006978569889777832) should be the same as actual (0.08723212362222289)

Aggregate x_HourOfDay and y_HourOfDay of rawFeature - dateFeature. 0.016493734169231777 was not less than 1.0E-10 expected aggregated LOCO value (0.016493734169231777) should be the same as actual (0.032987468338463555)
```

Reply:
@sanmitra

Reply:
@leahmcguire @gerashegalov The reason for tracking zero values is whenever we want to average LOCOs of a same raw text feature we are also including the zero values. 
E.g if text feature `TextA` has on a row 6 non zero values `loco1`, ..., `loco6` and 4 0s, we are dividing by 10 : 
(`loco1` + `loco2` + ... + `loco6`)/10

Reply:
LEt me write a fix that will not go over the zeros